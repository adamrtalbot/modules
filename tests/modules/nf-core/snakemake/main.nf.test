nextflow_process {

    name "Test Process SNAKEMAKE"
    script "modules/nf-core/snakemake/main.nf"
    process "SNAKEMAKE"
    tag "snakemake"

    test("Single-Read") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test', single_end:true ],
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assert process.success
            assert process.out.html.get(0).get(1) ==~ ".*/test_fastqc.html"
            assert path(process.out.html.get(0).get(1)).getText().contains("<tr><td>File type</td><td>Conventional base calls</td></tr>")
            assert process.out.zip.get(0).get(1) ==~ ".*/test_fastqc.zip"
        }

    }

}
