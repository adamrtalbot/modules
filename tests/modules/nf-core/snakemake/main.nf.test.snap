{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "input"
                        },
                        [
                            "2023-07-31T093650.969680.snakemake.log:md5,1ba1e5b1f45fd77c0a0b2fced5597c9a",
                            "aGVsbG8udHh0:md5,5d5b69e1a49e8d8aeac628ee5526e94e",
                            "hello.txt:md5,e59ff97941044f85df5297e1c302d260",
                            "versions.yml:md5,91f038cba572e2b7b6dfe06e0e088ce5"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "input"
                        },
                        [
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                "2023-07-31T093650.969680.snakemake.log:md5,1ba1e5b1f45fd77c0a0b2fced5597c9a"
                            ],
                            [
                                "aGVsbG8udHh0:md5,5d5b69e1a49e8d8aeac628ee5526e94e"
                            ],
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,91f038cba572e2b7b6dfe06e0e088ce5"
                ],
                "outputs": [
                    [
                        {
                            "id": "input"
                        },
                        [
                            "2023-07-31T093650.969680.snakemake.log:md5,1ba1e5b1f45fd77c0a0b2fced5597c9a",
                            "aGVsbG8udHh0:md5,5d5b69e1a49e8d8aeac628ee5526e94e",
                            "hello.txt:md5,e59ff97941044f85df5297e1c302d260",
                            "versions.yml:md5,91f038cba572e2b7b6dfe06e0e088ce5"
                        ]
                    ]
                ],
                "snakemake_dir": [
                    [
                        {
                            "id": "input"
                        },
                        [
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                
                            ],
                            [
                                "2023-07-31T093650.969680.snakemake.log:md5,1ba1e5b1f45fd77c0a0b2fced5597c9a"
                            ],
                            [
                                "aGVsbG8udHh0:md5,5d5b69e1a49e8d8aeac628ee5526e94e"
                            ],
                            [
                                
                            ],
                            [
                                
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,91f038cba572e2b7b6dfe06e0e088ce5"
                ]
            }
        ],
        "timestamp": "2023-07-31T09:36:55+0000"
    }
}